core/src/worker/workflow/managed_run.rs

```diff
@@ -1169,6 +1197,67 @@ impl ManagedRun {
     }
 }
 
+// Construct a new command sequence with query responses removed, and any
+// terminal responses removed, except for the first terminal response, which is
+// placed at the end. Return new command sequence, query commands, and a boolean
+// recording whether the result differs from the old behavior (i.e. whether
+// there were any non-terminal, non-query commands after the first terminal).
+// Note that multiple coroutines may have generated a terminal command, leading
+// to multiple terminal commands in the input to this function.
+fn preprocess_command_sequence(
+    mut commands: Vec<WFCommand>,
+) -> (Vec<WFCommand>, Vec<QueryResult>, bool) {
+    let mut query_results = vec![];
+    let mut terminals = vec![];
+    let (mut seen_terminal, mut any_moved) = (false, false);
+
+    commands = std::mem::take(&mut commands)
```

Comment:
`take` here and in the next fn no longer needed. (I guess it wasn't really before either, at some point that was a ref rather than being passed by value)